packages/sanity/src/core/releases/hooks/useFocusAwareFormState.ts

```diff
@@ -0,0 +1,81 @@
+import {type EditableReleaseDocument} from '@sanity/client'
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react'
+
+type ReleaseFormFields = Pick<EditableReleaseDocument['metadata'], 'title' | 'description'>
+
+interface UseFocusAwareFormStateOptions {
+  /** The external value from props/server */
+  externalValue: EditableReleaseDocument
+  id: string
+  extractData: (value: EditableReleaseDocument) => ReleaseFormFields
+}
+
+/**
+ * Hook for managing release form state that syncs with server data while respecting per-field focus.
+ *
+ * This prevents server updates from overriding user input during typing.
+ * Local state syncs with server data on a per-field basis only when the field is not focused.
+ *
+ * @internal
+ */
+export function useFocusAwareFormState({
+  externalValue,
+  id,
+  extractData,
+}: UseFocusAwareFormStateOptions) {
+  const incomingFormData = useMemo(() => extractData(externalValue), [externalValue, extractData])
+  const [localData, setLocalData] = useState(() => incomingFormData)
+  const [focusedField, setFocusedField] = useState<string | null>(null)
+  const previousIdRef = useRef(id)
+
+  const updateUnfocusedFields = useCallback(
+    (currentFormData: ReleaseFormFields) => {
+      const formFieldNames = Object.keys(incomingFormData) as Array<keyof ReleaseFormFields>
+      const unfocusedFieldUpdates = formFieldNames
+        .filter((field) => field !== focusedField)
+        .reduce<Partial<ReleaseFormFields>>((fieldUpdates, field) => {
+          fieldUpdates[field] = incomingFormData[field]
+
+          return fieldUpdates
+        }, {})
+
+      return {...currentFormData, ...unfocusedFieldUpdates}
+    },
+    [incomingFormData, focusedField],
+  )
+
+  useEffect(() => {
+    // New releases don't have _createdAt and should always sync to support local storage
+    const isEditingExistingRelease = Boolean(externalValue._createdAt)
+
+    // if tracking a new ID
+    if (previousIdRef.current !== id) {
+      previousIdRef.current = id
+      setLocalData(incomingFormData)
+      setFocusedField(null)
+    } else if (isEditingExistingRelease) {
+      setLocalData(updateUnfocusedFields)
+    } else {
+      setLocalData(incomingFormData)
```

Comment:
can probably be simplified to?
```suggestion
    } else {
    setLocalData(isEditingExistingRelease ? updateUnfocusedFields : incomingFormData)
```

or similar? Sorry the formatting isn't great